### PR TITLE
#25 rebased

### PR DESCRIPTION
**Pull request by dinosaure**
This PR is #25 with no conflict. First, I would like to know if I did not missed something if we compare #25 and this PR (/cc @djs55). Then, Last blocker to be able to merge this PR is to document more behavior of `read_line` as @hannesm asked.

I'm not an user of `mirage-channel`, however I completely understand purpose of this feature (even if I prefer to go on my way with this kind of problem with `ke`). So, I agree to merge and do a release then :+1:.

### CODE REVIEW

**Comment by dinosaure** [comment]
I add some tests to see what it happens in some examples from @hannesm and it seems that the `len` argument MUST count `CRLF` in any cases.

File: test/test_channel.ml [hunk 1]
```diff
@@ -67,6 +97,7 @@ let test_read_until_eof_then_write () =
 let suite = [
   "read_char + EOF" , `Quick, test_read_char_eof;
   "read_line"       , `Quick, test_read_line;
-  "read_exactly"    , `Quick, test_read_exactly;
-  "write after read EOF", `Quick, test_read_until_eof_then_write;
```
Comment (hannesm) [inline test/test_channel.ml]: why remove those tests?

**Comment by dinosaure** [comment]
One year after, I think that we can merge & release.

**Pull request merged**